### PR TITLE
(PUP-4936) Add files needed for smf puppet service

### PR DESCRIPTION
**Pull request by haus**
There were previously no smf files for puppet that weren't geared toward
opencsw. This commit adds the smf files from the PE service, and updates
them for current pathing.

### CODE REVIEW

File: ext/solaris/smf/puppet.xml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0"?>
```
Comment (stahnma) [inline ext/solaris/smf/puppet.xml]: Ow my :eyes:

File: ext/solaris/smf/puppet.xml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+<!DOCTYPE service_bundle SYSTEM "/usr/share/lib/xml/dtd/service_bundle.dtd.1">
+<!-- Original puppet manifest: Luke Kanies - puppetlabs.com -->
```
Comment (stahnma) [inline ext/solaris/smf/puppet.xml]: Is this the actual same one from Lak? I feel like other should get to put their names in here if he does.

**Comment by stahnma** [comment]
:+1:

File: ext/solaris/smf/puppet.xml (hunk 2 above)
Comment (haus) [inline ext/solaris/smf/puppet.xml]: Hah. This is from PE, so it has almost certainly seen some edits.

**Comment by geoffnichols** [comment]
:+1:

**Pull request merged**